warehouse/query-core/src/main/java/datawave/query/iterator/logic/AndIterator.java

```diff
@@ -350,11 +354,46 @@ public T move(T minimum) {
                 T transform = transformer.transform(next);
                 transforms.put(transform, next);
                 includeHeads.put(transform, itr);
+                
+                if (transform.compareTo(to) > 0) {
+                    to = transform;
+                }
             }
         }
         return includeHeads;
     }
     
+    protected TreeMultimap<T,NestedIterator<T>> moveIterators(SortedSet<T> toMove, T to) {
+        T highest = null;
+        
+        // get the highest key of toMove since it is likely to be the lowest cardinality
+        if (!toMove.isEmpty()) {
+            highest = toMove.last();
+        }
+        
+        while (highest != null && !includeHeads.isEmpty()) {
+            // if include heads is higher than the target key, adjust the target key to the highest since an intersection is not possible at anything but the
+            // highest
+            T highestInternal = includeHeads.keySet().last();
+            if (highestInternal.compareTo(to) > 0) {
+                to = highestInternal;
+            }
+            
+            // advance the iterators
+            includeHeads = moveIterators(highest, to);
```

Comment:
Are we risking a stack overflow here?

Reply:
this call is not on the set, but on a single value. This is not recursive.